Add Advanced field (Collectd plugin)

fprina opened the pull request:
* Add Advanced field
  Needed to append generic configurations to collecd.conf

* bump version

AdSchellevis commented:
I don't think we should add arbitrary input to config files, we try to avoid that as far as possible (some legacy configs have them unfortunately).
For custom hooks, it's better to provide template hooks like we use for squid. collectd probably also supports something like `Include "/etc/collectd.d/*.conf"` in its format.

fprina commented:
I've added in the form specific fields for this plug-in and removed the generic one
to be honest  do not like the format of the form but for the moment is the best i can provide ... maybe have a look

good idea the Include statement; which is the right location to put the directory in the OPNsense structure ?

mimugmail commented on net-mgmt/collectd/src/opnsense/mvc/app/models/OPNsense/Collectd/General.xml:
```diff
@@ -113,5 +113,29 @@
+            <default>0</default>
+            <Required>N</Required>
+        </p_openvpn_enable> 
+        <p_openvpn_statusfile type="TextField">
```
I would make this field Required Y and set a default, it saves time

fichtner replied:
isn't this the control socket ? if yes, we don't need a setting... we shouldn't support running / monitoring of stray openvpn instances here (openvpn not controlled by opnsense)

mimugmail replied:
No, control socket is in /var/etc/openvpn ... this is only a status

```
--status file [n]
Write operational status to file every n seconds.
Status can also be written to the syslog by sending a SIGUSR2 signal.

--status-version [n]
Choose the status file format version number. Currently n can be 1, 2, or 3 and defaults to 1.
```

mimugmail replied:
But it's perhaps not a bad idea to enable this by default with every server instance in the specific directory, then there's no need for this advanced hack :)

fichtner replied:
so the status file is added with the advanced options? ouch.. it really only makes sense to fill this here if we have proper core support for these two options

mimugmail commented on net-mgmt/collectd/src/opnsense/service/templates/OPNsense/Collectd/collectd.conf:
```diff
@@ -111,4 +111,14 @@ LoadPlugin write_graphite
 
+{% if helpers.exists('OPNsense.collectd.general.p_openvpn_enable') and OPNsense.collectd.general.p_openvpn_enable == '1' %}
+<Plugin openvpn>
+    StatusFile "{{ OPNsense.collectd.general.p_openvpn_statusfile }}"
```
If you make required Y in model this is ok without if clause

mimugmail commented on net-mgmt/collectd/src/opnsense/mvc/app/controllers/OPNsense/Collectd/forms/general.xml:
```diff
@@ -149,4 +149,40 @@
+        <id>general.p_openvpn_enable</id>
+        <label>Enable openvpn plugin</label>
+        <type>checkbox</type>
+        <help>The OpenVPN plugin reads the status file printed by OpenVP </help>
```
Small typo

mimugmail commented on net-mgmt/collectd/src/opnsense/mvc/app/controllers/OPNsense/Collectd/forms/general.xml:
```diff
@@ -149,4 +149,40 @@
+    </field>
+    <field>
+        <id>general.p_openvpn_improvednamingschema</id>
+        <label>openvpn Improved Naming Schema</label>
```
Please in label and help fields "OpenVPN" instead of "openvpn"

mimugmail commented on net-mgmt/collectd/src/opnsense/mvc/app/controllers/OPNsense/Collectd/forms/general.xml:
```diff
@@ -149,4 +149,40 @@
+    </field>
+    <field>
+        <id>general.p_openvpn_collectcompression</id>
+        <label>openvpn Collect Compression </label>
```
Remove last space

fichtner commented on net-mgmt/collectd/src/opnsense/mvc/app/models/OPNsense/Collectd/General.xml:
```diff
@@ -113,5 +113,29 @@
             <default>1</default>
```
^^^ model version number was not bumped, see above

fprina replied:
bumped to 1.1.0

fprina commented:
Sorry, I'm lost about the "the control socket " part

fichtner commented:
not to worry, we're having a debate about what this feature does and if we should add GUI support to the OpenVPN servers for the status file :)

fichtner commented:
@mimugmail can we pull this and / or do we need to adapt core ?

mimugmail commented:
If you give a okay for advanced fields I'll look at the code again. :)

fabianfrz commented:
@mimugmail will not look after this change. Don't wait for my approval because I don't know collectd.

fichtner commented:
Looking back, October 2017 is quite stale so we should reiterate as needed and close this for now.
